### PR TITLE
EDU-292: Angular Gen 2 None - Editable Regions

### PR DESCRIPTION
**Pull request by uttej-vsk**
## Description

This PR provides our documentation with Angular Gen 2 None code snippet for editable regions in custom components.

@samijaber Please review

### CODE REVIEW

File: packages/sdks/snippets/angular/src/app/editable-regions/editable-regions.component.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        [content]="content"
+        [model]="model"
+        [apiKey]="apiKey"
+        [customComponents]="customComponents"
+      />
+    </div>
+    <div *ngIf="notFound">404</div>
+  `,
+})
+export default class EditableRegionComponent implements OnInit {
```
Comment (sidmohanty11) [inline packages/sdks/snippets/angular/src/app/editable-regions/editable-regions.component.ts:29]: we can remove `implements OnInit`

File: packages/sdks/snippets/angular/src/app/app.module.ts [hunk 2]
```diff
@@ -12,10 +12,11 @@ import { BlogArticleComponent } from './blog-article/blog-article.component';
 import { CatchAllComponent } from './catch-all/catch-all.component';
 import { CustomChildComponent } from './custom-child/custom-child.component';
 import { CustomHeroComponent } from './custom-child/custom-hero/custom-hero.component';
+import { CustomColumnsComponent } from './editable-regions/custom-columns/custom-columns.component';
+import EditableRegionComponent from './editable-regions/editable-regions.component';
```
Comment (sidmohanty11) [inline packages/sdks/snippets/angular/src/app/app.module.ts:16]: can we use named export for `EditableRegionComponent` to maintain consistency?

File: packages/sdks/snippets/angular/src/app/editable-regions/editable-regions.component.ts (hunk 1 above)
Comment (uttej-vsk) [reply packages/sdks/snippets/angular/src/app/editable-regions/editable-regions.component.ts]: Ah! Though it works fine, after reading the Angular docs, one issue in [github](https://github.com/angular/angular/issues/5814) and other posts in Stackoverflow, it is more of the best practices. TS will help us with typos during compile time. 

If you also see the below images, if I implement **_onInit_**, when I mistype **_ngOnInit_**(), it throws an error. If we don't implement it, Angular compiles fine and gives no error, and then we will have difficulty figuring it out. 

Screenshot 2024-10-26 at 9 55 48 AM
Screenshot 2024-10-26 at 9 56 02 AM

That is the reason why it is there in the code. Let me know if it makes sense :)

File: packages/sdks/snippets/angular/src/app/app.module.ts (hunk 2 above)
Comment (uttej-vsk) [reply packages/sdks/snippets/angular/src/app/app.module.ts]: This is an overlook. Made the change and pushed the code.

File: packages/sdks/snippets/angular/src/app/editable-regions/editable-regions.component.ts (hunk 1 above)
Comment (sidmohanty11) [reply packages/sdks/snippets/angular/src/app/editable-regions/editable-regions.component.ts]: sure, do you think we should also add this to other components to maintain consistency?

**Review by sidmohanty11 — APPROVED** [review]

**Pull request merged**